scality_manila_utils/smb_helper.py

```diff
@@ -0,0 +1,359 @@
+        utils.fsync_path(root_export)
+
+        log.info("Deleting the export '%s' from the Samba registry",
+                 export_name)
+        process = subprocess.Popen(['net', 'conf', 'delshare', export_name],
+                                   stdout=subprocess.PIPE)
+        # `subprocess.communicate` returns a byte stream
+        output = text_type(process.communicate()[0])
+
+    if process.returncode != 0:
+        msg = ("Something went wrong while deleting the "
+               "export {0:s}: '{1:s}'".format(export_name, output))
+        if "SBC_ERR_NO_SUCH_SERVICE" in output:
+            log.warning(msg)
+        else:
+            raise EnvironmentError(msg)
+
+
+def _set_hosts_allow(export_name, hosts_allow):
+    """
+    Set the `hosts_allow` parameter for a given share.
+
+    :param export_name: name of export to grant access to
+    :type export_name: string (unicode)
+    :param hosts_allow: hosts allowed on this share
+    :type hosts_allow: iterable of `string`
+    """
+
+    cmd = ['net', 'conf', 'setparm', export_name,
+           '\"hosts allow\"', '\"%s\"' % ' '.join(hosts_allow)]
+    process = subprocess.Popen(' '.join(cmd), shell=True,
+                               stdout=subprocess.PIPE)
+    # `subprocess.communicate` returns a byte stream
+    output = text_type(process.communicate()[0])
+
+    if process.returncode != 0:
+        msg = ("Something went wrong while setting '{0!r}' as "
+               "the list of 'hosts allow' for share '{1:s}': {2:s}".format(
+                   hosts_allow, export_name, output))
+        raise EnvironmentError(msg)
+
+
+@ensure_environment
+@ensure_export_exists
+def grant_access(export_name, host, exports, *args, **kwargs):
+    """
+    Grant access for a host to an export.
+
+    :param export_name: name of export to grant access to
+    :type export_name: string (unicode)
+    :param host: host to grant access for
+    :type host: string (unicode)
+    :param exports: all the defined shares in the Samba registry
+    :type exports: dictionary
+    """
+
+    hosts_allow = exports[export_name]['hosts allow'].split()
+
+    if host in hosts_allow:
+        msg = "Host '{0:s}' already allowed on share '{1:s}'".format(
+            host, export_name)
+        raise ClientExistsException(msg)
+
+    hosts_allow.append(host)
+
+    with utils.elevated_privileges():
+        _set_hosts_allow(export_name, hosts_allow)
+
+
+@ensure_environment
+@ensure_export_exists
+def revoke_access(export_name, host, exports, *args, **kwargs):
+    """
+    Revoke access for a host to an export.
+
+    :param export_name: name of export for revocation
+    :type export_name: string (unicode)
+    :param host: host to revoke access for
+    :type host: string (unicode)
+    :param exports: all the defined shares in the Samba registry
+    :type exports: dictionary
+    """
+
+    hosts_allow = exports[export_name]['hosts allow'].split()
+
+    try:
+        hosts_allow.remove(host)
```

Comment:
What's the rationale behind this construct? Why not just `if host not in hosts_allow` ...